### ISSUE TITLE
The  behavior_path_planner can not plan the right path in some positions as the target

### ISSUE THREAD

**Issue by shulanbushangshu** [autowarefoundation/autoware.universe#2310]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

When I set the different positions of target in the lane,the behavior_path_planner can not plan the right path in some positions.
case 1:

![lane_follow_normal](https://user-images.githubusercontent.com/102840938/202373400-c4fa7907-9b87-444b-962c-3652d2fd65ab.png)

case 2:

![Screenshot from 2022-11-16 16-27-25](https://user-images.githubusercontent.com/102840938/202373503-010ee0ac-ee9c-43d4-9314-09614c48402b.png)

case 3:
![Screenshot from 2022-11-17 14-35-22](https://user-images.githubusercontent.com/102840938/202374493-e4c7f533-d48c-4635-b272-c5c68621420e.png)

case 4:

![Screenshot from 2022-11-17 14-37-25](https://user-images.githubusercontent.com/102840938/202374591-db59890e-5f00-4547-a91d-fec04fabd627.png)

### Expected behavior

The behavior_path_planner shall plan the right path in all positions of the lanes.

### Actual behavior

The behavior_path_planner can not plan the right path when I set some different positions  as the target in the lanes.

### Steps to reproduce

1.Start autoware planning simulator;
2. Set initial position 
3. Set target in the different positions in the lane(far from the initial position)

### Versions

1.OS: Ubuntun 20.04
2.ROS2: Galactic
3.Autoware: Autoware.universe:main

**Comment by ismetatabay** [comment]
Hi, can you please share map (pcd and lanelet) files? The lanelets which highlighted as I have shown below picture show that the route has been successfully generated by the mission_planner, I didn't see any problem in this case. If you're worried about not seeing the green path in rviz, the path will move forward as the vehicle move.  The parameter that is ```forward_path_length``` limits the path . You can also increase the generated path by changing this parameter.

https://github.com/autowarefoundation/autoware.universe/blob/29e8b435a4416e5db0eb6c7da0c4b0457e6c529b/launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml#L4

![Selection_013](https://user-images.githubusercontent.com/56237550/202441455-706b5a08-a704-494c-a113-3df04c369f02.png)

**Comment by shulanbushangshu** [comment]
@ismetatabay  As you say, the route has been successfully generated by the mission_planner.Yes,I get what you mean.The distance from the target is more than 300m.But  the problem is that final path of the behavior_path_planner is short(I print the size of the path and it shows that there are just  two points in the path)

**Comment by shulanbushangshu** [comment]
@ismetatabay  I  test the code and show the information about the path of behavior_path_planner as below:

case1:

![Screenshot from 2022-11-18 12-55-42](https://user-images.githubusercontent.com/102840938/202620971-71dcc9e7-16c4-42d0-9cff-dab0cbf6aa84.png)



case 2:

![Screenshot from 2022-11-18 12-56-37](https://user-images.githubusercontent.com/102840938/202621080-ca99ec5e-c051-4aa1-9043-89f84ab8ada4.png)

**Comment by ismetatabay** [comment]
@shulanbushangshu This is weird thing that it shouldn't happen. At this case, if ```forward_path_length``` is 300m and ```backward_path_length``` is 5m, the number of path points created should be approximately 305/2 ≅152. To see if this situation is related to distance goal distance , I modified the sample .osm file of autoware as in the image below. 

I've tried many different scenarios but haven't run into a problem. As expected, if the distance to the goal was greater than 300 meters, approximately 152 path points were generated on ```bpp```.

Is it possible for you to share the lanelet2 .osm file?

![Selection_015](https://user-images.githubusercontent.com/56237550/202711451-9ea3f161-f69b-44a4-96bf-71f7360eb03a.png)
